cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/LowResourceUtilizationTest.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.analyzer;
+
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuUsageDistributionGoal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskUsageDistributionGoal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundUsageDistributionGoal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundUsageDistributionGoal;
+import com.linkedin.kafka.cruisecontrol.analyzer.goals.ResourceDistributionGoal;
+import com.linkedin.kafka.cruisecontrol.common.DeterministicCluster;
+import com.linkedin.kafka.cruisecontrol.common.Resource;
+import com.linkedin.kafka.cruisecontrol.config.constants.AnalyzerConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.ExecutorConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.MonitorConfig;
+import com.linkedin.kafka.cruisecontrol.exception.OptimizationFailureException;
+import com.linkedin.kafka.cruisecontrol.executor.ExecutionProposal;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+import com.linkedin.kafka.cruisecontrol.model.ReplicaPlacementInfo;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Unit test for testing low utilization threshold taking effect to prevent rebalance for resource distribution goals
+ */
+@RunWith(Parameterized.class)
+public class LowResourceUtilizationTest {
+
+  private static final Map<Resource, Double> BROKER_CAPACITY = new HashMap<Resource, Double>() {{
```

Comment:
There would be a syntax error `cannot use "<>" with anonymous inner classes` if we do not specify types in `<>` in this form

Reply:
This is supported in Java 9 and above [source](https://java2blog.com/java-9-anonymous-inner-classes-and-diamond-operator/). Since, CC is compiled with Java 11, diamond notation should work.

@Lincong I assume you were building with Java 8? I am ok to leave it as is to maintain build compatibility with Java 8.